wallet/src/main/java/com/android/identity_credential/wallet/CardViewModel.kt

```diff
@@ -0,0 +1,215 @@
+package com.android.identity_credential.wallet
+
+import android.content.Context
+import android.graphics.BitmapFactory
+import androidx.compose.runtime.mutableStateListOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.identity.credential.Credential
+import com.android.identity.credential.CredentialStore
+import com.android.identity.credentialtype.CredentialTypeRepository
+import com.android.identity.issuance.CredentialCondition
+import com.android.identity.issuance.CredentialExtensions.credentialConfiguration
+import com.android.identity.issuance.CredentialExtensions.credentialIdentifier
+import com.android.identity.issuance.CredentialExtensions.housekeeping
+import com.android.identity.issuance.CredentialExtensions.issuingAuthorityIdentifier
+import com.android.identity.issuance.CredentialExtensions.state
+import com.android.identity.issuance.IssuingAuthority
+import com.android.identity.issuance.IssuingAuthorityRepository
+import com.android.identity.securearea.SecureAreaRepository
+import com.android.identity.util.Logger
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Instant
+
+class CardViewModel : ViewModel() {
+
+    val cards = mutableStateListOf<Card>()
+
+    private lateinit var context: Context
+    private lateinit var credentialStore: CredentialStore
+    private lateinit var issuingAuthorityRepository: IssuingAuthorityRepository
+    private lateinit var secureAreaRepository: SecureAreaRepository
+    private lateinit var credentialTypeRepository: CredentialTypeRepository
+
+    fun getCard(cardId: String): Card? {
+        for (card in cards) {
+            if (card.id == cardId) {
+                return card
+            }
+        }
+        return null
+    }
+
+    fun refreshCard(card: Card) {
+        val credential = credentialStore.lookupCredential(card.id)
+        if (credential == null) {
+            Logger.w(TAG, "No credential with id ${card.id}")
+            return
+        }
+        viewModelScope.launch(Dispatchers.IO) {
+            credential.housekeeping(
+                issuingAuthorityRepository,
+                secureAreaRepository,
+                true,
+            )
+        }
+    }
+
+    fun deleteCard(card: Card) {
```

Comment:
I am a bit surprised that APIs for both CardViewModel and CredentialStore are synchronous... It makes more sense for CardViewModel to basically be an optimistic synchronous state. But CredentialStore, I'd think should expose state which is consistent with what is in the storage? But then should not it modification APIs be asynchronous (suspend)?

Reply:
I'm not sure what you mean by "APIs for both CardViewModel and CredentialStore are synchronous" or what you are suggesting. Please elaborate.

Reply:
Let's revisit it separately, I do not think this has to be figured out as a part of this PR.